### PR TITLE
Make convert parallelism controllable

### PR DESCRIPTION
**Pull request by totakke**
* Makes the number of threads of `cljam.algo.convert/convert` changeable
* Adds `--thread` options to `convert` subcommand
* Adds benchmark of `cljam.algo.convert/convert`

### CODE REVIEW

File: src/cljam/algo/convert.clj [hunk 1]
```diff
@@ -11,20 +13,24 @@
...
-                                         (encoder/encode-alignment bb % refs)
-                                         {:data (.array bb)})
-                                      chunk))
-                              (partition-all num-block (sam/read-alignments rdr {})))]
-        (sam/write-blocks wtr blocks)))))
+  (let [refs (sam-util/make-refs hdr)
+        n-threads (get-exec-n-threads)
+        map-fn (if (= n-threads 1)
+                 map
+                 (partial cp/pmap (cp/threadpool (dec n-threads))))]
```
Comment (alumi) [inline src/cljam/algo/convert.clj]: I noticed that claypoole takes care of [switching map/pmap](https://github.com/TheClimateCorporation/claypoole/blob/9a37bfd312742ddd48b7c146c5e5bb9e924d1a52/src/clj/com/climate/claypoole.clj#L368) and [creating a temporal thread pool](https://github.com/TheClimateCorporation/claypoole/blob/9a37bfd312742ddd48b7c146c5e5bb9e924d1a52/src/clj/com/climate/claypoole/impl.clj#L227).

So it's equivalent to

```clojure
(partial cp/pmap (if (= n-threads 1) :serial (dec n-threads)))
```

**Comment by alumi** [comment]
Thank you for the patch!
LGTM other than a minor comment. Please check.

**Comment by totakke** [comment]
You're right. I've just fixed it.

**Comment by alumi** [comment]
Merged. Thanks :+1:

**Comment by totakke** [comment]
Thank you :)

**Pull request merged**